lib/matplotlib/style/core.py

```diff
@@ -84,6 +85,73 @@ def use(style):
                 raise IOError(msg % style)
 
 
+def get_substyles(styles):
```

Comment:
> As you know, the matplotlib.style.use function takes a string or a list

Right, but the description in this docstring suggests that `'main'` is the only style passed to `style.use` and then *that* stylesheet specifies `'a'`, `'b'`, and `'c'`. If I understand this PR correctly, that branching is not possible. (FWIW, I'd lean toward *not* supporting that sort of thing, but it's not that big of a deal, either way.)

I think the idea behind this is valuable, and would help support a major use case: Setting a default style sheet in `.matplotlibrc`. That said, I would think the hierarchy of style settings would be reversed: The user would specify the desired style sheet (e.g. `'matplotlib-classic'`, `matplotlib-2.0`) and then any other setting in the default `.matplotlibrc` would be applied on top of that style sheet. In that case, the `'style'` parameter would specify a parent style sheet, rather than a child.

In any case, I think this specific change deserves its own PR and should be thoroughly tested. Personally, I think the more focused a PR is, the better.